frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/WhenExpressionForm.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useFormikContext } from 'formik';
+import { Grid, GridItem, TextInputTypes } from '@patternfly/react-core';
+import { InputField, DropdownField, TextColumnField } from '@console/shared';
+import AutoCompletePopover from '../../shared/common/auto-complete/AutoCompletePopover';
+import { PipelineBuilderFormikValues } from '../pipeline-builder/types';
+import { WhenExpressionOperatorType } from '../pipeline-builder/const';
+
+type WhenExpressionFormProps = {
+  namePrefix?: string;
+  autoCompleteValues: string[];
+};
+
+const WhenExpressionForm: React.FC<WhenExpressionFormProps> = ({
+  namePrefix,
+  autoCompleteValues,
+}) => {
+  const { setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
+  const { t } = useTranslation();
+
+  return (
+    <Grid hasGutter>
```

Comment:
Hmmm, think this is a weird use of grid... but it works... so maybe it's not an issue. Would think a Flex split and a div (or nothing) for the `TextColumnField`.

Not the end of the world to leave it this way.

Reply:
Will refactor this while creating the custom field array component for when expression form in ODC-5800.